tensorflow/compiler/xla/literal.cc

```diff
@@ -1725,6 +1726,16 @@ StatusOr<Literal> LiteralBase::BitcastConvert(const Shape& dest_shape) const {
   Literal out(dest_shape);
   std::memcpy(out.root_piece_.buffer(), root_piece().buffer(),
               root_piece().size_bytes());
+
+  if (!tensorflow::port::kLittleEndian) {
```

Comment:
We can't add complexity for little endian support everywhere in XLA without a holistic plan, would you like to start a discussion on one?

Reply:
Sure, I've started a discussion in https://github.com/openxla/xla/discussions/5 . Could you please take a look? Thanks! @Cheshire @sherhut

Reply:
Hi @cheshire  ,

The Bitcast related code change is a bit different from other endianness handling code. It's related to the Bitcast op itself. This won't be fixed by the generic endianness solution anyway.

Could you please review it again? Thank you!

Reply:
I understand, but I'm very hesitant to take a big-endianness complexity everywhere in the code without a clear path forward on how to get big-endian support , design and estimate of how much extra code it will bring

Reply:
Thanks @cheshire ,

We're sorting out the list of endianness related scenarios in TensorFlow/XLA code we've seen so far and will post it at https://github.com/openxla/xla/discussions/5. We can discuss more on adding generic big-endian support there.

Reply:
Can you add a comment with explanation. This is obvious now but might be less obvious later.

Something along the lines of "perform the reshape on little endian encoding even on big endian machines" would already help I think.

Also, please format the code.

Otherwise I agree, there is no general way of doing this.

Reply:
Thanks @sherhut !

I've added the comment and formatted the code. Could you please take a look?